enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,545 @@
+- No concrete migration plan from legacy to OLM v1.
+- Coordination around external team dependencies.
+- The BundleDeployment API has a large net of permissions.
+- The upstream rukpak project has a direct dependency on cert-manager.
+
+#### Mitigations
+
+The v1alpha1 CRDs won't be deployed by default, and those APIs will be hidden behind the "TechPreviewNoUpgrade" feature set that admins need to explicitly enable before cluster creation. Any new components introduced during these early phases will also specify the `release.openshift.io/feature-gate` annotation in their CVO manifests, and set that key's value to "TechPreviewNoUpgrade". This annotation will ensure that the CVO won't deploy any tech preview components.
+
+Instructing admins to configure upfront RBAC to limit the BundleDeployment API to the minimum set of users may help mitigate the plain provisioner's wide net of permissions. The process for constructing this upfront RBAC may require explicit documentation.
+
+We can mitigate the upstream rukpak project's direct dependency on the cert-manager operator by removing that dependency in favor of the downstream service-ca-operator component.
+
+### Drawbacks
+
+- Implementing this PO mechanism using legacy OLM would provide more immediate business value. That implementation is outlined below in the alternatives section.
+- Early phases will be delivered through tech preview guidelines. This leads to little customer feedback, given it precludes production clusters.
+- Early phases may introduce a new OLM 1.x control plane that has little-to-no interaction with the existing, legacy OLM 0.x control plane.
+- Early phases may introduce validating webhooks. These webhooks can impact the cluster's overall health when they're not available.
+- Early phases may introduce validating webhooks. By default, these webhooks will be deployed using an HA configuration, but may not have logic to accommodate SNO cluster environments.
+- Early phases may introduce a dynamic watching implementation for managing the underlying OLM operator resources. This has potential performance implications that need to be measured through benchmarking and/or addressed before any GA phase.
+- Future phase designs may require a large-scale rewrite of the OLM component.
+
+## Design Details
+
+### Open Questions [optional]
+
+- For components that are commonly used, but would become optional under this plan, what would the experience be for mirroring them into a disconnected environment? If I mirror a release image in the future, will that include less stuff than today, and I'll have to take extra steps to mirror the optional components I want?
+- What does migration look like for a component that's removed from the payload and becomes optional, or one that's OLM-based today and would fall under this new category?
+- Does phase 0 support OLM-based operators that use OLM's management of the APIService and webhook resource?
```

Comment:
@joelanford Do you remember whether there was a technical limitation for managing the "APIServiceDefinitions" or "WebhookDefinitions" for phase 0, or did we have those initial safeguards to provide less friction when building out the registry+v1 conversion logic?

Reply:
IIRC, I think we said that for now we do not support registry+v1 bundles that have webhook and/or apiservice definitions because those both make use of OLM's built-in certificate generation.

I don't think rukpak should be actively managing certificates for workloads it manages, so I'm not sure that's a feature we should be carrying forward. I think the better solution is to have those bundles declare a dependency on cert-manager and then include the necessary cert-manager objects.

Reply:
Okay, I'll update the requirements for this phase, and add some language around the certification generation issue and handling this (potentially) in phase 1 when dependency resolution comes into play.

Reply:
Updated in the last commit.